rest_framework/filters.py

```diff
@@ -79,6 +79,9 @@ def must_call_distinct(self, queryset, search_fields):
                 search_field = search_field[1:]
             parts = search_field.split(LOOKUP_SEP)
             for part in parts:
+                if isinstance(queryset, models.QuerySet) and part in queryset.query.annotations:
+                    # This field is annotated
+                    continue
```

Comment:
I need to look into this further, but I think the annotation check needs to occur before the loop, and on the whole `search_field`, not the individual parts. 

- You can't annotate across joins. e.g., parsing `annotated` from `some_rel__annotated` is invalid. 
- You can't join across annotations. e.g., parsing `some_rel` from `annotated__some_rel` is invalid.

Code should probably be...

```python
for search_field in search_fields:
    opts = queryset.model._meta
    if search_field[0] in self.lookup_prefixes:
        search_field = search_field[1:]
    # Annotated fields do not need to be distinct
    if isinstance(queryset, models.QuerySet) and search_field in queryset.query.annotations:
         return False
    parts = search_field.split(LOOKUP_SEP)
    for part in parts:
        ...
```